api/src/main/java/org/openmrs/OrderSet.java

```diff
@@ -0,0 +1,139 @@
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+/**
+ * @since 1.12
+ * Represents the grouping of orders into a set,
+ * so as to give decision support for the doctors
+ */
+public class OrderSet extends BaseOpenmrsMetadata implements Serializable {
+	
+	public static final long serialVersionUID = 72232L;
+	
+	/**
+	 * Restrictions put on saving an orderSet.
+	 * ALL: All the members of the orderSet need to be selected for saving
+	 * ONE: Only one of the member of the orderSet needs to be selected for saving
+	 * ANY: Any of the members of the orderSet can be selected for saving
+	 */
+	public enum Operator {
+		ALL, ONE, ANY
+	};
+	
+	private Integer orderSetId;
+	
+	private Operator operator;
+	
+	private List<OrderSetMember> orderSetMembers;
+	
+	/**
+	 * Gets the orderSetId
+	 *
+	 * @return the orderSetId
+	 */
+	public Integer getOrderSetId() {
+		return orderSetId;
+	}
+	
+	/**
+	 * Sets the orderSetId
+	 *
+	 * @param orderSetId the orderSetId to set
+	 */
+	public void setOrderSetId(Integer orderSetId) {
+		this.orderSetId = orderSetId;
+	}
+	
+	/**
+	 * Gets the operator
+	 *
+	 * @return the operator
+	 */
+	public Operator getOperator() {
+		return operator;
+	}
+	
+	/**
+	 * Sets the operator
+	 *
+	 * @param operator the operator to set
+	 */
+	public void setOperator(Operator operator) {
+		this.operator = operator;
+	}
+	
+	/**
+	 * Gets the orderSetMembers
+	 *
+	 * @return the orderSetMembers
+	 */
+	public List<OrderSetMember> getOrderSetMembers() {
+		if(orderSetMembers == null) {
+			orderSetMembers = new ArrayList<OrderSetMember>();
+		}
+		return orderSetMembers;
+	}
+	
+	/**
+	 * Sets the orderSetMembers
+	 *
+	 * @param orderSetMembers the orderSetMembers to set
+	 */
+	public void setOrderSetMembers(List<OrderSetMember> orderSetMembers) {
+		for (OrderSetMember orderSetMember : orderSetMembers) {
+			addOrderSetMember(orderSetMember);
+		}
+	}
+	
+	/**
+	 * Adds an orderSetMember to the existing list of orderSetMembers
+	 * 
+	 * @param orderSetMember the new orderSetMember to be added
+	 * @param position the position where it is to be added, if position is null it adds to the last position 
+	 */
+	
+	public void addOrderSetMember(OrderSetMember orderSetMember, Integer position) {
+		if (orderSetMember == null || getOrderSetMembers().contains(orderSetMember)) return;
+		position = position == null ? getOrderSetMembers().size() : position - 1;
+		orderSetMember.setOrderSet(this);
+		getOrderSetMembers().add(position, orderSetMember);
```

Comment:
Allowing negative position to insert from the end of the list would be a nice feature – i.e., `addOrderSetMember(osm, -1)` would add to end of the list and `addOrderSetMember(osm,-2)` would insert into the second to last position.

Note: if we make changes to this method please make the same changes to `OrderGroup.addOrder(Order, Int)`, since these two methods should have consistent behavior.